torchvision/datasets/utils.py

```diff
@@ -224,15 +223,25 @@ def download_file_from_google_drive(file_id: str, root: str, filename: Optional[
             params = {'id': file_id, 'confirm': token}
             response = session.get(url, params=params, stream=True)
 
-        if _quota_exceeded(response):
+        # Ideally, one would use response.status_code to check for quota limits, but google drive is not consistent
+        # with their own API, refer https://github.com/pytorch/vision/issues/2992#issuecomment-730614517.
+        # Should this be fixed at some place in future, one could refactor the following to no longer rely on decoding
+        # the first_chunk of the payload
+        response_content_generator = response.iter_content(32768)
```

Comment:
Does it matter that the chunksize used to be `chunk_size=128` for this specific check and now it's 32768?

Reply:
I think I tested it back then and 128 was enough to get to the important part of the message. But since we use the chunk anyway, any reasonable chunk size should be ok.

Reply:
As far as I could debug manually, google drive either returns:

1. A html stating that the quota is exceeded. We currently parse the contents of this html in `_quota_exceeded`, and yes, size of 128 is sufficient to get to the phrase/part we check against here. Larger chunk sizes didn't hurt in my debugging.
2. The desired payload, e.g. a tarball of a dataset (which cannot be decoded and parsed).

The reason parsing the html is worse than using the return_code is that if google decides to change the formatting of the returned html (or has an edge case where it returns a **different** html to signal something else) our check will fail, we will write html to disk again, and subsequently fail to unpack it in the next step, leaving the users with a cryptic error message.

We could be more strict here by rejecting all responses that have a payload which's first chunk can be decoded to strings, but the best long term solution would probably be for google to fix/adhere to their API.